Fail to compile return with match{} and nested function call

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relvant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.2.2 f18adf7.5d653a3
**OS:** linux, Linux version 5.10.8-arch1-1 (linux@archlinux)

**What did you do?**

```
struct Data {
	array []int
}

fn (d Data) len() int {
	return d.array.len
}

fn make_result() []Data {
	return []
}

fn f_doesnotcompile(d Data) []Data {
	return match d.len() {
		1 { make_result() }
		else { make_result() }
	}
}

fn f_compiles1(d Data) []Data {
	return match d.array.len {
		1 { make_result() }
		else { make_result() }
	}
}

fn f_compiles2(d Data) []Data {
	length := d.array.len
	return match length {
		1 { make_result() }
		else { make_result() }
	}
}

fn main() {
	println(f_doesnotcompile({}))
	println(f_compiles1({}))
	println(f_compiles2({}))
}
```


**What did you expect to see?**

A compiled program.
 
**What did you see instead?**

```
==================
/tmp/v/issue-return_match.9577516100469669949.tmp.c:8620: error: '_t67' undeclared
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.
[...]
```

The relevant bit in the generated c:
```
VV_LOCAL_SYMBOL Array_main__Data main__f_doesnotcompile(main__Data d) {
		Array_main__Data _t68;
	
	if (_t67 == 1) {
		_t68 = main__make_result();
	}
	else {
		_t68 = main__make_result();
	}	int _t67 = main__Data_len(d); 
		return _t68;
}
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -3544,14 +3544,15 @@ fn (mut g Gen) match_expr(node ast.MatchExpr) {
 		cond_var = g.new_tmp_var()
 		g.write('${g.typ(node.cond_type)} $cond_var = ')
 		g.expr(node.cond)
-		g.writeln('; ')
+		g.writeln(';')
+		g.stmt_path_pos << g.out.len
 		g.write(line)
 	}
 	if need_tmp_var {
 		g.empty_line = true
 		cur_line = g.go_before_stmt(0)
 		tmp_var = g.new_tmp_var()
-		g.writeln('\t${g.typ(node.return_type)} $tmp_var;')
+		g.writeln('${g.typ(node.return_type)} $tmp_var;')
 	}
 
 	if is_expr && !need_tmp_var {

--- a/vlib/v/tests/match_in_fn_call_test.v
+++ b/vlib/v/tests/match_in_fn_call_test.v
@@ -1,0 +1,42 @@
+struct Data {
+	array []int
+}
+
+fn (d Data) len() int {
+	return d.array.len
+}
+
+fn make_result() []Data {
+	return []
+}
+
+fn f_doesnotcompile(d Data) []Data {
+	return match d.len() {
+		1 { make_result() }
+		else { make_result() }
+	}
+}
+
+fn f_compiles1(d Data) []Data {
+	return match d.array.len {
+		1 { make_result() }
+		else { make_result() }
+	}
+}
+
+fn f_compiles2(d Data) []Data {
+	length := d.array.len
+	return match length {
+		1 { make_result() }
+		else { make_result() }
+	}
+}
+
+fn test_match_in_fn_call() {
+	println(f_doesnotcompile({}))
+	assert f_doesnotcompile({}) == []Data{}
+	println(f_compiles1({}))
+	assert f_compiles1({}) == []Data{}
+	println(f_compiles2({}))
+	assert f_compiles2({}) == []Data{}
+}

